libavformat/hls.c

```diff
@@ -2289,8 +2310,22 @@ static int hls_read_packet(AVFormatContext *s, AVPacket *pkt)
         }
 
         /* Segment metadata */
-        av_dict_set_int(&metadata_dict, "segNumber", pls->cur_seq_no, 0);
-        relative_seq_no = pls->cur_seq_no - pls->start_seq_no;
+        {
+            int cur_seq_no = pls->cur_seq_no;
+            /* If the playlist is VOD then let's cap it to the number of segments */
+            if (pls->finished) {
+                if (pkt->pos >= pls->segment_boundary_position) {
+                        pls->reported_segment_number++;
+                    pls->segment_boundary_position += pls->segments[pls->reported_segment_number]->actual_size;
```

Comment:
do we need a check to make sure pls->reported_segment_number is not >= number of segments?